analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -101,10 +90,28 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
             context.State[symbol] is null || !context.State[symbol].HasConstraint<ObjectConstraint>();
     }
 
-    private static bool IsParameterDereferenced(IOperationWrapperSonar operation) =>
-        operation.Parent is { } parent
-        && (parent.IsAnyKind(DereferenceOperations)
-            || (parent.Kind == OperationKindEx.Conversion && IsParameterDereferenced(parent.ToSonar())));
+    private static IOperation NullDereferenceCandidate(IOperation operation)
+    {
+        var candidate = operation.Kind switch
+        {
+            OperationKindEx.Invocation => operation.ToInvocation().Instance,
+            OperationKindEx.FieldReference => operation.ToFieldReference().Instance,
+            OperationKindEx.PropertyReference => operation.ToPropertyReference().Instance,
+            OperationKindEx.EventReference => operation.ToEventReference().Instance,
+            OperationKindEx.Await => operation.ToAwait().Operation,
+            OperationKindEx.ArrayElementReference => operation.ToArrayElementReference().ArrayReference,
+            _ => null,
+        };
+
+        return candidate?.Kind == OperationKindEx.Conversion
+            ? ConversionOperand(candidate)
+            : candidate;
+    }
+
+    private static IOperation ConversionOperand(IOperation operation) =>
+        operation.Kind == OperationKindEx.Conversion
+            ? ConversionOperand(operation.ToConversion().Operand)
+            : operation;
```

Comment:
```suggestion
        return UnwrapConversions(candidate);
    }

    private static IOperation UnwrapConversions(IOperation operation) =>
        operation?.Kind == OperationKindEx.Conversion
            ? UnwrapConversions(operation.ToConversion().Operand)
            : operation;
```

Reply:
I'd rather keep the name `ConversionOperand`, as it shows what the return value will be.

Reply:
The issue I see with that is that it will only return a conversion operand if the operation is a conversion.